<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CreateStrategyDialog</class>
 <widget class="QDialog" name="CreateStrategyDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>550</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create Strategy Instance</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
    QDialog { background-color: #ffffff; color: #1f1f1f; font-family: 'Segoe UI'; }
    QLabel { background-color: transparent; color: #333333; font-weight: 600; font-size: 12px; border: none; }
    QLineEdit, QSpinBox, QDoubleSpinBox, QTextEdit, QComboBox {
        background-color: #ffffff; color: #333333;
        border: 1px solid #d1d5db; border-radius: 4px; padding: 6px;
        font-size: 12px;
    }
    QLineEdit:focus, QComboBox:focus, QSpinBox:focus, QDoubleSpinBox:focus, QTextEdit:focus {
        border: 1px solid #2563eb;
    }
    QPushButton {
        background-color: #f3f4f6; color: #1f2937;
        border: 1px solid #d1d5db; border-radius: 4px;
        padding: 6px 16px; font-weight: 600; font-size: 12px;
    }
    QPushButton:hover { background-color: #e5e7eb; border-color: #9ca3af; }
    QPushButton:pressed { background-color: #d1d5db; }
    QTextEdit { font-family: 'Consolas', monospace; font-size: 11px; }
   </string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_name">
       <property name="text">
        <string>Instance Name</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="m_nameEdit"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_type">
       <property name="text">
        <string>Strategy Type</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="m_typeCombo"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_symbol">
       <property name="text">
        <string>Symbol</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="m_symbolEdit"/>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_account">
       <property name="text">
        <string>Account ID</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="m_accountEdit"/>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_segment">
       <property name="text">
        <string>Segment</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QComboBox" name="m_segmentCombo">
       <item>
        <property name="text">
         <string>NSECM</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>NSEFO</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BSECM</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BSEFO</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_stopLoss">
       <property name="text">
        <string>Stop Loss</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QDoubleSpinBox" name="m_stopLossSpin">
       <property name="decimals">
        <number>2</number>
       </property>
       <property name="maximum">
        <double>1000000000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_target">
       <property name="text">
        <string>Target</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QDoubleSpinBox" name="m_targetSpin">
       <property name="decimals">
        <number>2</number>
       </property>
       <property name="maximum">
        <double>1000000000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="label_entryPrice">
       <property name="text">
        <string>Entry Price</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QDoubleSpinBox" name="m_entryPriceSpin">
       <property name="decimals">
        <number>2</number>
       </property>
       <property name="maximum">
        <double>1000000000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="label_qty">
       <property name="text">
        <string>Quantity</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QSpinBox" name="m_qtySpin">
       <property name="maximum">
        <number>1000000</number>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="label_params">
       <property name="text">
        <string>Parameters</string>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <widget class="QTextEdit" name="m_paramsEdit">
       <property name="placeholderText">
        <string>Optional JSON parameters. Example: {&quot;threshold&quot;: 0.5}</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="text">
        <string>Create</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>okButton</sender>
   <signal>clicked()</signal>
   <receiver>CreateStrategyDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>CreateStrategyDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
